xGame/client/src/Components/Match/MatchBoard.js

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from "react";
+import Grid from "@material-ui/core/Grid";
+import PropTypes from "prop-types";
+import {
+  findLocation,
+  createBoardPiece,
+  showTable,
+} from "./HelpfulMatchTools.js";
+
+import "./MatchStyle.css";
+
+const MatchBoard = (props) => {
+  const boardState = props.boardState;
+  const activeColor = props.activeColor;
+  const [chessBoard, setChessBoard] = useState([]);
+  const [piecesOnBoard, setPiecesOnBoard] = useState([]);
+  const [piecesToUpdate, setPiecesToUpdate] = useState({ add: [], remove: [] });
+  const [selected, setSelected] = useState([]);
+
+  const checkForAdditions = () => {
+    let newUpdateList = piecesToUpdate;
+    for (let x = 0; x < boardState.length; x++) {
+      let matchFound = false;
+      for (let y = 0; y < piecesOnBoard.length; y++) {
+        if (boardState[x] === piecesOnBoard[y]) {
+          matchFound = true;
+        }
+      }
+      if (!matchFound) {
+        newUpdateList.add.push(boardState[x]);
+      }
+    }
+    setPiecesToUpdate(newUpdateList);
+  };
+
+  const checkForRemoval = () => {
+    let newUpdateList = piecesToUpdate;
+    for (let x = 0; x < piecesOnBoard.length; x++) {
+      let matchFound = false;
+      for (let y = 0; y < boardState.length; y++) {
+        if (boardState[x] === piecesOnBoard[y]) {
+          matchFound = true;
+        }
+      }
+      if (!matchFound) {
+        newUpdateList.remove.push(boardState[x]);
+      }
+    }
+    setPiecesToUpdate(newUpdateList);
+  };
+
+  //change to alter boardState
+  const addSelected = (selectedPieceName, selectedLocation) => {
+    console.log("clicked");
+    let tempSelected = selected;
+    tempSelected.push({
+      pieceName: selectedPieceName,
+      location: selectedLocation,
+    });
+    if (tempSelected.length === 2) {
+      console.log("switch");
+      setSelected([]);
+    } else {
+      setSelected(tempSelected);
+    }
```

Comment:
If I'm understanding this correctly, you should be able to remove these lines and instead do
setSelected({
      pieceName: selectedPieceName,
      location: selectedLocation,
    });

Reply:
I am mainly planning on totally replacing this with the http request, but until then yeah you're right